pagefaultgames/pokerogue
# [QoL] Pokemon Info Container always on top of Battle UI

## What are the changes?
When catching a pokemon, the Pokemon Info Container is not displayed on top of all of the battle UI, mainly the enemy modifier bar and held items are overlapping with the Info Container. See Videos for before vs after. The depth change has no effect on the Egg hatching phase, which can be seen in video 3.

## Why am I doing these changes?
This is neither visually pleasing nor intended. And I've found no other PR fixing this issue. 

## What did change?
- increase depth of the parentContainer of the InfoContainer to 3 (above scene.uiContainer) on show()
- save initial depth to `initialParentDepth` on show()
- return to initial depth on hide()

### Screenshots/Videos

**1. BEFORE**

https://github.com/pagefaultgames/pokerogue/assets/1838682/e31aa360-abcd-4bf3-a7d0-0457550c2ddc


**2. AFTER**

https://github.com/pagefaultgames/pokerogue/assets/1838682/e1e37a87-979c-437e-9f55-f30f104cbe8f


**3. EGG HATCHING PHASE (NO EFFECT)**

https://github.com/pagefaultgames/pokerogue/assets/1838682/b08add73-cf86-48ae-98bf-764052089b2c


## How to test the changes?
Overrides:
Give you a master ball:
`export const POKEBALL_OVERRIDE: { active: boolean, pokeballs: PokeballCounts } = {
  active: true,
  pokeballs: {
    [PokeballType.POKEBALL]: 5,
    [PokeballType.GREAT_BALL]: 0,
    [PokeballType.ULTRA_BALL]: 0,
    [PokeballType.ROGUE_BALL]: 0,
    [PokeballType.MASTER_BALL]: 1,
  }
};`

Give the enemy a held item:
`export const OPP_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [{name: "BERRY", count: 1, type: BerryType.SITRUS}];`

Start a new run and catch the first pokemon with the master ball. The Info Container will open and is displayed ABOVE the held berry.

## Checklist
